### PR TITLE
feat: [DEL-5140]: delegate delete apis

### PR DESCRIPTION
**Pull request by arpit3107**
## Describe your changes
These are customer facing automation apis needed to delete delegates.

Scenarios tested:
1.Existing flow of delegate deletion from UI is working.
2. Deleted delegate with api-keys using these newly introduced apis.
3. Tried to delete connected delegates, but it threw exception as expected.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)

### CODE REVIEW

**Comment by raghuAtWings** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 129 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFNAyEJJT5PWqITd-:-NFFFNAz-6Nmb_91KtUG:b-7gsq2) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L129)):*
> ```Java
> 
>   @DELETE
>   @Path("groups/{groupIdentifier}")
> ```

don't put groups

/delegate-setup/{delegateIdentifier} should be good. User doesn't know anything about group

**Comment by raghuAtWings** [comment]
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 95 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFhbJAIMmMfz0MvXz:-NFFFhbK0i3XqyZZMaPS:b-2guby3) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L95)):*
> ```Java
> 
>   @DELETE
>   @Path("groups/{groupIdentifier}")
> ```

same here

**Comment by raghuAtWings** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 2439 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFlWv3xKLjIaC9gZ4:-NFFFlWv3xKLjIaC9gZ5:bxuw8ir) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L2439)):*
> ```Java
> 
>   @Override
>   public void deleteDelegateGroupV2(String accountId, String orgId, String projectId, String identifier) {
> ```

who uses this? if not used remove

**Review by raghuAtWings — COMMENTED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: 0 of 1 approvals obtained.

File: 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java [hunk 1]
```diff
@@ -588,12 +588,13 @@ public RestResponse<DelegateApprovalResponse> rejectDelegatesUsingToken(
   @ExceptionMetered
   @AuthRule(permissionType = ACCOUNT_MANAGEMENT)
   @AuthRule(permissionType = MANAGE_DELEGATES)
-  public RestResponse<Void> delete(
+  @ApiKeyAuthorized(permissionType = MANAGE_DELEGATES)
+  public RestResponse<String> delete(
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java:592]: Generally returning string response is not good as it blocks future development

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: 8 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java` line 129 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFNAyEJJT5PWqITd-:-NFIgWp53ZGPz8gLwEmY:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateSetupNgResource.java#L129)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

don't put groups

/delegate-setup/{delegateIdentifier} should be good. User doesn't know anything about group
</blockquote></details>

Done.

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: 8 file(s) to review. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 95 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFhbJAIMmMfz0MvXz:-NFIgYkx2kr3BhUlzE5C:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L95)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java` line 592 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFE8zNjEC5faXPZ6rHt-r1-592:-NFIgZjo6DRJXsQcE4lF:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java#L592)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

Generally returning string response is not good as it blocks future development
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 2439 at r1](https://reviewable.io/reviews/harness/harness-core/39186#-NFFFlWv3xKLjIaC9gZ4:-NFIg_wa31aZ-BgGm9Cw:b-sy7p12) ([raw file](https://github.com/harness/harness-core/blob/3046e656c8f80d1d4960db130f1714de0d782e26/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L2439)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

who uses this? if not used remove
</blockquote></details>

its internally used at a lot of places.

**Review by gaurav-jain1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: 8 file(s) to review. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java` line 95 at r2](https://reviewable.io/reviews/harness/harness-core/39186#-NFIwXJK-Yd94ih7SmBa:-NFIwXJL7XlA9KT6PZkE:b-d3hoee) ([raw file](https://github.com/harness/harness-core/blob/005250a7cfbbbdd6b6dab23dcec6ca2a20b174ab/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgSetupInternalResource.java#L95)):*
> ```Java
> 
>   @DELETE
>   @Path("delete-delegate")
> ```

you can skip mentioning `delete` here, as api method type is delete, which is self explanatory



_Code quote:_
```Java
delete-delegate
```

**Review by raghuAtWings — DISMISSED** [review]
> Reviewed 6 of 8 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: 1. unresolved discussions.

**Review by gaurav-jain1 — DISMISSED** [review]
> Reviewed 1 of 7 files at r1, 6 of 8 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: :shipit: complete! ✓

**Review by AnupamIO — APPROVED** [review]

**Review by AnupamIO — APPROVED** [review]
> Reviewed 1 of 7 files at r1, 6 of 8 files at r2, 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: :shipit: complete! ✓

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39186)* status: :shipit: complete! ✓

**Pull request merged**